lib/enum/nested.rb

```diff
@@ -18,21 +18,30 @@ def initialize(people_array)
   end
 
   def ages_sum
+    @people.inject(0){ |sum, person| sum + person.age }
   end
 
   def average_age
+    ages_sum / @people.count
   end
 
   def total_years_programming_experience_for_all_languages
+    # @people.inject(0) do |sum, person|
+    # sum += person.years_language_experience.inject(0){ |sum, language| sum += @years_language_experience[language] }
+    # end
+    @people.flat_map{ |person| person.years_language_experience }.inject(0){ |sum, years| sum + years }
```

Comment:
You're on the right track! Here's one solution: 

`people
      .map(&:years_language_experience)
      .map(&:values)
      .flatten
      .inject(:+)`